src/main/java/io/openliberty/tools/intellij/lsp4mp/lsp4ij/operations/codeactions/LSPCodeActionAnnotator.java

```diff
@@ -127,29 +127,31 @@ protected Info doCollectInformationStep(PsiFile file, Editor editor) {
 
     private Info doAnnotateStep(Info collectedInfo) {
         Collection<CompletableFuture<?>> futures = new ArrayList<>();
-        for(Map.Entry<LanguageServerWrapper, Collection<ItemInfo>> entry : collectedInfo.itemInfos.entrySet()) {
-            if (supportsCodeAction(entry.getKey())) {
-                for(ItemInfo itemInfo : entry.getValue()) {
-                    Diagnostic diagnostic = (Diagnostic) itemInfo.highlighter.getErrorStripeTooltip();
-                    CodeActionContext context = new CodeActionContext(Collections.singletonList(diagnostic));
-                    CodeActionParams params = new CodeActionParams();
-                    params.setContext(context);
-                    params.setTextDocument(new TextDocumentIdentifier(LSPIJUtils.toUri(collectedInfo.file).toString()));
-                    params.setRange(diagnostic.getRange());
-                    CompletableFuture<List<Either<Command, CodeAction>>> codeAction = entry.getKey().getInitializedServer().thenComposeAsync(server -> {
-                        return server.getTextDocumentService().codeAction(params);
-                    });
-                    futures.add(codeAction);
-                    codeAction.thenAcceptAsync(actions -> {
-                        itemInfo.actions = actions;}
-                    );
+        if (collectedInfo != null) {
```

Comment:
We do not need to call this method if the input is null because `futures` will remain empty. I think this method should be handled the same as `doApplyStep()` at lines 176-179.

Reply:
Made the suggested changes.